python/chronos/src/bigdl/chronos/data/experimental/xshards_tsdataset.py

```diff
@@ -271,6 +271,85 @@ def roll(self,
                                                         feature_col, target_col)
         return self
 
+    def scale(self,
+              scaler,
+              fit=True):
+        '''
+        Scale the time series dataset's feature column and target column.
+
+        :param scaler: sklearn scaler instance or dictionary, StandardScaler, MaxAbsScaler,
+               MinMaxScaler and RobustScaler are supported. If fit is true, it should be
+               sklearn scaler instance; if fit is false, scaler should be a dictionary
+               whose key is id and value is sklearn scaler instance.
+        :param fit: if we need to fit the scaler. Typically, the value should
+               be set to True for training set, while False for validation and
+               test set. The value is defaulted to True.
+
+        :return: the xshardtsdataset instance.
+
+        Assume there is a training set tsdata and a test set tsdata_test.
+        scale() should be called first on training set with default value fit=True,
+        then be called on test set with the same scaler and fit=False.
+
+        >>> from sklearn.preprocessing import StandardScaler
+        >>> scaler = StandardScaler()
+        >>> tsdata.scale(scaler, fit=True)
+        >>> tsdata_test.scale(scaler, fit=False)
+        '''
+        def _fit(df, id_col, scaler, feature_col, target_col):
+            df[feature_col + target_col] = scaler.fit(df[feature_col + target_col])
+            return {id_col: df[id_col][0], "scaler": scaler}
+
+        def _transform(df, id_col, scaler, feature_col, target_col):
+            from sklearn.utils.validation import check_is_fitted
+            from bigdl.nano.utils.log4Error import invalidInputError
+            try:
+                invalidInputError(not check_is_fitted(scaler[df[id_col][0]]),
+                                  "scaler is not fittedd")
+            except Exception:
+                invalidInputError(False,
+                                  "When calling scale for the first time, "
+                                  "you need to set fit=True.")
```

Comment:
well I think you can just write one `invalidInputError(not check_is_fitted(scaler[df[id_col][0]]), "xxx")`